pkg/jobWorker/namespaces/cgroup.go

```diff
@@ -0,0 +1,107 @@
+package namespaces
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+)
+
+const (
+	CPU_WEIGHT_File  = "cpu.weight"
+	MEMORY_HIGH_File = "memory.high"
+	IO_WEIGHT_File   = "io.weight"
+
+	/*
+		Common Permission Usages
+
+		0755 Commonly used on web servers. The owner can read, write, execute. Everyone else can read and execute but not modify the file.
+
+		0777 Everyone can read write and execute. On a web server, it is not advisable to use ‘777’ permission for your files and folders, as it allows anyone to add malicious code to your server.
+
+		0644 Only the owner can read and write. Everyone else can only read. No one can execute the file.
+
+		0655 Only the owner can read and write, but not execute the file. Everyone else can read and execute, but cannot modify the file.
+	*/
+	FILE_MODE_EVERYONE = 0777
+	FILE_MODE_WEB      = 0755 //0o500 = 0o500
+)
+
+var (
+	rootCgroupPath = "/sys/fs/cgroup"
+)
+
+// AddProcess mutates the given cmd to instruct to add the PID of the started process to a given cgroup
+func AddProcess(cgroupName string, cmd *exec.Cmd) error {
+	// Add job's process to cgroup
+	cgroupDir := GetCGroupPath(cgroupName)
+	fd, err := syscall.Open(filepath.Join(cgroupDir, "cgroup.procs"), os.O_RDWR, 0)
+	if err != nil {
+		return err
+	}
+	// This is where clone args and namespaces for user, PID and fs can be set
+	cmd.SysProcAttr.CgroupFD = fd
+
+	return nil
+}
+
+// CreateCGroup creates a directory in the cgroup root path to signal cgroup to create a group
+// TODO in production we could check here the cgroup was created correctly, such as checking cgroup.controllers file for supported controllers
+func CreateCGroup(cgroupName string) (err error) {
+	cgroupDir := GetCGroupPath(cgroupName)
+
+	// create a directory structure like /sys/fs/cgroup/<uuid>
+	log.Printf("create cgroup:%s", cgroupDir)
+	if err := os.Mkdir(cgroupDir, FILE_MODE_EVERYONE); err != nil {
```

Comment:
why `FILE_MODE_EVERYONE` instead of `FILE_MODE_WEB`?

Reply:
@greedy52 changes to FILE_MODE_WEB. 

`Explanation:` I've got 'permission denied' error during my cgroup testing and tried to create it with 'everyone read and write' permission, but I've changes to  FILE_MODE_WEB, retest and look ok for me.